docs/index.rst

```diff
@@ -7,6 +7,7 @@ Qiskit Terra documentation
   :hidden:
 
   API References <apidocs/terra>
+  Explanation <explanation/index>
```

Comment:
How finalized is the plan for Terra docs to migrate to the folder structure representing https://diataxis.fr? It's a little awkward to add a new top level section in the docs with only one page. I want to make sure that there would be a plan to move more content here - avoid a situation where in 1 year we still have this top level section with only a single page.

Reply:
primitives explanation: https://github.com/Qiskit/qiskit-terra/issues/10188